dart/lib/flat_buffers.dart

```diff
@@ -676,18 +676,48 @@ class Builder {
   }
 
   int _writeString(String value) {
-    // TODO(scheglov) optimize for ASCII strings
-    List<int> bytes = utf8.encode(value);
-    int length = bytes.length;
+    // [utf8.encode()] is slow (up to at least Dart SDK 2.13). If the given
+    // string is ASCII we can just write it directly, without any conversion.
+    final originalTail = _tail;
+    if (!_tryWriteASCIIString(value)) {
```

Comment:
This will be pretty bad for non-Latin use cases, as well as many non-English use cases.

Also - how does your hand made method compare to using `ascii.encode`?

Reply:
(and then catching an exception if it fails)

Reply:
I did not but I really don't see how it could be faster because it has to allocated a buffer for a copy of the string and this has to be then copied once more - so there's one more allocation and one more copy than my hand-rolled aproached does

Reply:
See here: from `lib/convert/ascii.dart`

```dart
  Uint8List convert(String string, [int start = 0, int? end]) {
    var stringLength = string.length;
    end = RangeError.checkValidRange(start, end, stringLength);
    // TODO(38725): Remove workaround when assignment promotion is implemented
    if (end == null) {
      throw RangeError("Invalid range");
    }
    var length = end - start;
    var result = Uint8List(length);
    for (var i = 0; i < length; i++) {
      var codeUnit = string.codeUnitAt(start + i);
      if ((codeUnit & ~_subsetMask) != 0) {
        throw ArgumentError.value(
            string, "string", "Contains invalid characters.");
      }
      result[i] = codeUnit;
    }
    return result;
  }
  ```

Reply:
I've changed my code to use the subset mask (`~0x7F`) which is marginally faster than the `>= 128` comparison I had before

Reply:
> This will be pretty bad for non-Latin use cases, as well as many non-English use cases.

what about the suggestion from the PR description: "Maybe this should be something configurable, either on the builder level or field level..."

Reply:
That would probably be fine but I think it should be defaulted to utf8.

Reply:
I've added an argument to optionally enable this optimization